modules/shared/src/main/scala/io/constellationnetwork/schema/swap.scala

```diff
@@ -161,12 +161,17 @@ object swap {
   @derive(decoder, encoder, order, show)
   case class SpendTransaction(
     fee: SpendTransactionFee,
-    lastValidOrdinal: SnapshotOrdinal,
+    validUntilOrdinal: SnapshotOrdinal,
     allowSpendRef: Hash,
-    currency: Option[CurrencyId],
+    currencyId: Option[CurrencyId],
     amount: SwapAmount
   )
 
+  object SpendTransaction {
+    implicit val ordering: Ordering[SpendTransaction] =
```

Comment:
It shouldn't be an implicit ordering. Because of that line, two different allow spends with the same validUntilOrdinal would be considered equal. Ordering on partial fields should be defined per use case

Reply:
👍🏼 
However, I need the 
`@derive(..., ordering, ...)`
otherwise it will fail the SortedSet

Reply:
no, I think you don't. You need it indeed in a place you create a sorted set, but not globally

Reply:
and, it will be used to deduplicate entries. so keep in mind that "two different allow spends with the same validUntilOrdinal would be considered equal"